### ISSUE TITLE
Using DerivedCombiningClass.txt to determine width is inappropriate

### ISSUE THREAD

**Issue by philipc** [jquast/wcwidth#10]
DerivedCombiningClass.txt contains the Canonical_Combining_Class field from UnicodeData.txt (see http://www.unicode.org/reports/tr44/#Canonical_Combining_Class_Values). This field is intended to be used for the collation algorithm.

wcwidth.py is currently assuming that characters are zero width combining characters if and only if they have a non-zero combining class. I think this is an invalid assumption. For example, characters that are enclosing marks (General Category = Me) all have a zero combining class, but they are also zero width combining characters.

I'm not sure what the standard way to determine zero width combining characters is. One possibility is to check for a General Category of Mn or Me, but I don't know if there are any exceptions to this. Also note that there are combining characters that do have a width (category Mc).

**Comment by jquast** [comment]
> I'm not sure what the standard way to determine zero width combining characters is.

Me neither, any help appreciated, or if you prefer such caveats documented in greater detail in the README, feel free to pull request and add to such section, https://github.com/jquast/wcwidth/blob/master/README.rst#todo or rename it 'caveats'

**Comment by jquast** [comment]
Please see related tickets https://github.com/jquast/wcwidth/issues/2 and PR https://github.com/jquast/wcwidth/pull/5

**Comment by philipc** [comment]
I think that checking for a general category of Mn or Me is closer to correct than using the canonical combining class, so I've started working on that change.

One issue I've encountered so far is the behavior of returning -1 for combining characters. I think -1 is better reserved for format characters that have an undefined width. Combining characters should return their actual width (0 or 1). This would be consistent with the behavior of http://www.cl.cam.ac.uk/~mgk25/ucs/wcwidth.c
